src/store/initial-state.js

```diff
@@ -262,10 +250,14 @@ const getInitialState = (props = {}) => {
     urlParams
   );
 
-  return {
+  const mergedInitialState = {
     ...nonPipelineState,
     ...pipelineState,
   };
+
+  return props.props
+    ? deepmerge(mergedInitialState, props.props)
+    : mergedInitialState;
```

Comment:
```suggestion
  const initialState = {
    ...nonPipelineState,
    ...pipelineState,
  };

  return props.props
    ? deepmerge(initialState, props.props)
    : initialState;
```

Reply:
this is a nit in terms of naming, because I feel `mergedInitialState` is used when deepmerge is done but that wasn't the case.